### PR TITLE
created get user by id endpoint

### PR DESCRIPTION
**Pull request by ojcorrigan**
created a get user by username end point, tested and working, tested for 404 as all inputs are being treated as a sting so was the only one I could think would work

ticket #20

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -55,395 +55,425 @@ describe('getTopics', () => {
...
+          username: 'rogersop',
+          name: 'paul',
+          avatar_url:
+            'https://avatars2.githubusercontent.com/u/24394918?s=400&v=4',
+        });
+      });
+  });
+  test('/api/user/i_dont_exist responds 404 not found', () => {
+    return request(app)
+      .get('/api/i_dont_exist')
```
Comment (karlaferry) [inline __tests__/app.test.js]: Your test description says `api/user/i_dont_exist` but your get request is `api/i_dont_exist`.

File: models/users-models.js [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+
 exports.selectUsers = () => {
   return db.query('SELECT username FROM users').then((users) => {
     return users.rows;
   });
 };
+
+exports.selectUserById = (username) => {
+  return db
+    .query('SELECT username, name, avatar_url FROM users WHERE username = $1', [
```
Comment (karlaferry) [inline models/users-models.js]: Don't forget semicolons 😄  Also the users table only has these three keys anyway so you can just say `SELECT *...`

**Review by karlaferry — APPROVED** [review]
Not bad!! Just left some minor comments but this is approved 😄 Make sure to change the test as per my comment though!

File: models/users-models.js (hunk 2 above)
Comment (ojcorrigan) [reply models/users-models.js]: I have gone through all of my models now!

**Pull request merged**